src/services/review/RepoCheckerService.ts

```diff
@@ -439,28 +437,32 @@ export default class RepoCheckerService {
 
           return this.checker(repo)
             .andThen((errors) => this.reporter(repo, errors))
-            .map<RepoErrorDto>((errors) => ({
+            .map<BrokenLinkErrorDto>((errors) => ({
               status: "success",
               errors,
             }))
 
-            .andThen((errors) =>
-              this.remover(repo, repoExists).andThen(() => ok(errors))
-            )
+            .andThen((errors) => {
+              if (repoExists) {
+                return ok(errors)
+              }
+              // since this repo was cloned during this operation, we should remove it
+              return this.remover(repo).andThen(() => ok(errors))
+            })
             .andThen((errors) =>
               this.deleteLock(repo).andThen(() => ok(errors))
             )
         })
       )
       .orElse((error) => {
         if (error.status === "loading") {
-          return okAsync<RepoErrorDto>({
+          return okAsync<BrokenLinkErrorDto>({
             status: "loading",
           })
         }
         this.deleteLock(repo)
         logger.error(`Error running site checker for repo ${repo}: ${error}`)
-        return errAsync<RepoErrorDto, RepoErrorDto>({
+        return errAsync<BrokenLinkErrorDto, BrokenLinkErrorDto>({
```

Comment:
honestly, this is my only gripe atm - returning an error **in both success and error** states is very confusing.

there should eb some differences between them, no?

Reply:
calling this directly at the router level, and fe expect error dto mah. 

how would you structure this ah? cuz its either this or at the router call site, we need to define the shape which feels like a weird place to construct the dto